packages/core/src/sync-historical/service.ts

```diff
@@ -620,48 +625,115 @@ export class HistoricalSyncService extends Emittery<HistoricalSyncEvents> {
     const logIntervals = this.buildLogIntervals({ fromBlock, toBlock, logs });
 
     for (const logInterval of logIntervals) {
-      const { startBlock, endBlock, logs, transactionHashes } = logInterval;
-      (this.blockCallbacks[endBlock] ||= []).push(async (block) => {
-        await this.syncStore.insertLogFilterInterval({
+      const { startBlock, endBlock } = logInterval;
+
+      if (this.blockCallbacks[endBlock] === undefined)
+        this.blockCallbacks[endBlock] = [];
+
+      this.blockCallbacks[endBlock].push(async (block) => {
+        await this._insertLogFilterInterval({
+          logInterval,
+          logFilter: logFilter.criteria,
           chainId: logFilter.chainId,
           block,
+        });
+
+        this.common.metrics.ponder_historical_completed_blocks.inc(
+          { network: this.network.name, contract: logFilter.contractName },
+          endBlock - startBlock + 1,
+        );
+      });
+    }
+
+    this.logFilterProgressTrackers[logFilter.id].addCompletedInterval([
+      fromBlock,
+      toBlock,
+    ]);
+
+    this.enqueueBlockTasks();
+
+    this.common.logger.trace({
+      service: "historical",
+      msg: `Completed LOG_FILTER task adding ${logIntervals.length} BLOCK tasks [${fromBlock}, ${toBlock}] (contract=${logFilter.contractName}, network=${this.network.name})`,
+    });
+  };
+
+  private factoryLogFilterTaskWorker = async ({
+    factory,
+    fromBlock,
+    toBlock,
+  }: FactoryLogFilterTask) => {
+    const iterator = this.syncStore.getFactoryChildAddresses({
+      chainId: factory.chainId,
+      factory: factory.criteria,
+      upToBlockNumber: BigInt(toBlock),
+    });
+
+    // Note: Is this creating too many database queries at once?
+    const childAddresses: Address[][] = [];
+    for await (const childContractAddressBatch of iterator) {
+      childAddresses.push(childContractAddressBatch);
+    }
+
+    const logs = await Promise.all(
+      childAddresses.map(async (c) =>
+        this._eth_getLogs({
+          address: c,
+          topics: factory.criteria.topics,
+          fromBlock: numberToHex(fromBlock),
+          toBlock: numberToHex(toBlock),
+        }),
+      ),
+    ).then((l) => l.flat());
```

Comment:
I wonder if this should be properly batched, where we query a batch of addresses from the DB, make an eth_getLogs call, add a block callback, then repeat.

Reply:
Sounds good, I was a little weary of this spot which is why I left the note